core-bundle/src/Resources/contao/forms/Form.php

```diff
@@ -259,6 +259,30 @@ protected function compile()
 			$this->processFormData($arrSubmitted, $arrLabels, $arrFields);
 		}
 
+		// Remove any uploads, if form did not validate (#1185)
+		if ($doNotSubmit && $hasUpload && !empty($_SESSION['FILES']))
```

Comment:
Do we not need the `&& !Environment::get('isAjaxRequest')` check here as well?

Reply:
I don't think so. It looks like a bug to me that `$_SESSION['FILES']` was not purged, when the form was sent via AJAX. Or conversely: it should not matter whether the form was sent via AJAX or not. If the form does not validate, any uploaded file should be deleted, regardless of the type of request, should it not?